ocaml/typing/typetexp.ml

```diff
@@ -470,9 +476,15 @@ and transl_type_aux env policy mode styp =
       let rec loop acc_mode args =
         match args with
         | (l, arg_mode, arg) :: rest ->
-          let l = transl_label l in
+          let l = transl_label l (Some arg) in
           check_arg_type arg;
-          let arg_cty = transl_type env policy arg_mode arg in
+          let arg_cty =
+            match arg.ptyp_desc with
+            | Ptyp_extension ({txt="src_pos"; _}, _payload) ->
```

Comment:
This duplicates logic from within `transl_label`. Maybe `transl_label` should return both the translated label and also the translated argument type, which would avoid this.